### PR TITLE
Cache metrics configuration

### PR DESCRIPTION
**Pull request by sasabolic**
Enable metrics for the specified list of cache names.

### CODE REVIEW

File: memcached-spring-boot-autoconfigure/build.gradle [hunk 1]
```diff
@@ -15,7 +15,9 @@ dependencies {
     optionalImplementation 'org.springframework.boot:spring-boot-actuator'
     optionalImplementation 'org.springframework.cloud:spring-cloud-context'
     optionalImplementation 'io.micrometer:micrometer-core'
+    implementation 'org.springframework:spring-context-support'
 
+    testImplementation 'javax.cache:cache-api:1.1.1'
```
Comment (igorbolic) [inline memcached-spring-boot-autoconfigure/build.gradle:20]: You might not need to specify the version i.e. it should be resolved from the Spring Boot Dependency pom file

File: memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCacheManagerTest.java [hunk 2]
```diff
@@ -85,4 +90,62 @@ public void whenGetDisabledCacheThenIncreaseCacheSize() {
...
     }
+
+	@Test
+	public void whenGetDisabledCacheThenNoOpCache() {
+        MemcachedCacheProperties.CacheConfig cacheConfig = new MemcachedCacheProperties.CacheConfig();
+        cacheConfig.setExpiration(Duration.ofHours(3));
+        cacheConfig.setMetricsEnabled(true);
+        cacheConfig.setDisabled(true);
+
+        Map<String, MemcachedCacheProperties.CacheConfig> configurationPerCache = Stream.of(new Object[][] {
```
Comment (igorbolic) [inline memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCacheManagerTest.java:101]: Couldn't you use the `Collections.singletonMap()`

File: memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCachePropertiesConfigPerCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+
+    @Test
+    public void whenGetServers_thenCorrectValue() {
+        List<InetSocketAddress> result = memcachedCacheProperties.getServers();
+
+        assertThat(result).isNotNull();
+        assertThat(result.isEmpty()).isFalse();
+        assertThat(result.size()).isEqualTo(2);
+        // @formatter:off
```
Comment (igorbolic) [inline memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCachePropertiesConfigPerCacheTest.java:60]: Can you remove `@fomatter:off/on`

**Review by igorbolic — COMMENTED** [review]
Left a few comments... could you also update the README with the new configuration options.

File: memcached-spring-boot-autoconfigure/build.gradle [hunk 4]
```diff
@@ -15,7 +15,9 @@ dependencies {
     optionalImplementation 'org.springframework.boot:spring-boot-actuator'
     optionalImplementation 'org.springframework.cloud:spring-cloud-context'
     optionalImplementation 'io.micrometer:micrometer-core'
+    implementation 'org.springframework:spring-context-support'
```
Comment (igorbolic) [inline memcached-spring-boot-autoconfigure/build.gradle:18]: Could you also group `implementation` dependencies together

Comment (sasabolic) [reply memcached-spring-boot-autoconfigure/build.gradle:18]: Sure

File: memcached-spring-boot-autoconfigure/build.gradle (hunk 1 above)
Comment (sasabolic) [reply memcached-spring-boot-autoconfigure/build.gradle:20]: Good catch!

File: memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCacheManagerTest.java (hunk 2 above)
Comment (sasabolic) [reply memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCacheManagerTest.java:101]: True, I'll change it

File: memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCachePropertiesConfigPerCacheTest.java (hunk 3 above)
Comment (sasabolic) [reply memcached-spring-boot-autoconfigure/src/test/java/io/sixhours/memcached/cache/MemcachedCachePropertiesConfigPerCacheTest.java:60]: True, will be removed

**Review by igorbolic — APPROVED** [review]
Looks good 👍

**Pull request merged**